src/Tribe/Query.php

```diff
@@ -719,6 +687,86 @@ public static function posts_orderby( $order_sql, $query ) {
 		}
 
 		/**
+		 * Adds a custom SQL join when ordering by venue or organizer is desired.
+		 *
+		 * @param string   $join_sql
+		 * @param wp_query $query
+		 *
+		 * @return string
+		 */
+		public static function posts_join_venue_organizer( $join_sql, $query ) {
+			global $wpdb;
+
+			switch ( $query->get( 'orderby' ) ) {
+				case 'venue':
+					$join_sql .= " LEFT JOIN {$wpdb->postmeta} tribe_order_by_venue_meta ON {$wpdb->posts}.ID = tribe_order_by_venue_meta.post_id AND tribe_order_by_venue_meta.meta_key='_EventVenueID' LEFT JOIN {$wpdb->posts} tribe_order_by_venue ON tribe_order_by_venue_meta.meta_value = tribe_order_by_venue.ID ";
+					break;
+				case 'organizer':
+					$join_sql .= " LEFT JOIN {$wpdb->postmeta} tribe_order_by_organizer_meta ON {$wpdb->posts}.ID = tribe_order_by_organizer_meta.post_id AND tribe_order_by_organizer_meta.meta_key='_EventOrganizerID' LEFT JOIN {$wpdb->posts} tribe_order_by_organizer ON tribe_order_by_organizer_meta.meta_value = tribe_order_by_organizer.ID ";
+					break;
+				default:
+					return $join_sql;
+					break;
+			}
+
+			/**
+			 * Ensures we add the matching corresponding code to the order clause.
+			 *
+			 * We use posts_clauses (as opposed to posts_orderby) in this case to avoid
+			 * it being overwritten by Tribe__Events__Admin_List methods.
+			 *
+			 * @see Tribe__Events__Admin_List
+			 */
+			add_filter( 'posts_clauses', array( __CLASS__, 'posts_orderby_venue_organizer' ), 100, 2 );
+
+			/**
+			 * @deprecated since 4.0.2
+			 */
+			$join_sql = apply_filters( 'tribe_events_query_posts_join_orderby', $join_sql );
```

Comment:
Yeah I was swithering about doing just that.

In the end I was reluctant to make it a 'noisy' deprecation because it feels like a hook name we may wish to restore and re-use in a context that isn't venue/organizer specific.

Happy to add it though if you think it's worth it?